bindings/python/tests/test_parse.py

```diff
@@ -32,3 +32,60 @@ def get_parsed_info(tmp_path, file_contents):
     )
 
     return parsed_info
+
+
+def test_class_template(tmp_path):
+    file_contents = """
+    template <typename T>
+    struct AStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    class_template = parsed_info["members"][0]
+
+    assert class_template["kind"] == "CLASS_TEMPLATE"
+    assert class_template["name"] == "AStruct"
+
+
+def test_template_non_type_parameter(tmp_path):
+    file_contents = """
+    template <int N>
+    struct AStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    class_template = parsed_info["members"][0]
+    template_non_type_parameter = class_template["members"][0]
+
+    assert template_non_type_parameter["kind"] == "TEMPLATE_NON_TYPE_PARAMETER"
+    assert template_non_type_parameter["element_type"] == "Int"
+    assert template_non_type_parameter["name"] == "N"
+
+
+def test_function_template(tmp_path):
+    file_contents = """
+    template <typename T>
+    void aFunction() {}
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    function_template = parsed_info["members"][0]
+
+    assert function_template["kind"] == "FUNCTION_TEMPLATE"
+    assert function_template["result_type"] == "void"
+    assert function_template["name"] == "aFunction"
+
+
+def test_template_type_parameter(tmp_path):
```

Comment:
This one misses testing for function templates. I'd go for integrating this in the respective tests

Reply:
Since the rest of the suite is one test per cursor kind, you could also add a free template function to this file_contents and assert the same assertions. I'm not married to the idea of 1 kind 1 test, though, and it might be more cohesive, but you open the door to joining some tests in #19.

Reply:
Added a function template in this one.